timeseries/src/autogluon/timeseries/models/local/statsforecast.py

```diff
@@ -391,3 +415,343 @@ def _get_model_type(self):
+
+    This implementation uses the ``ceil((1 - alpha) * (n + 1)) / n`` quantile of nonconformity scores
+    as the critical value. By definition, it follows that for small alpha and small n, this may result
+    in "infinite" forecast intervals. In this case, the absolute residual quantiles are clipped to the
+    sample maximum. This is why this method can be expected to undercover when the calibration window
+    is short and the required alpha is small (forecasts for very low or high quantiles are required).
+    """
+
+    max_num_conformalization_windows = 5
+
+    def _get_nonconformity_scores(
+        self,
+        time_series: pd.Series,
+        local_model_args: Dict,
+    ) -> np.ndarray:
+        h = self.prediction_length
+        y = time_series.values.ravel()
+
+        if len(y) <= h:
+            # if there is only prediction_length many time steps in sample, we fall back to
+            # the naive-1 forecaster to compute residuals for as many time steps as possible
+            nonconf_scores = np.abs(y - y[0])
+            if len(y) > 1:
+                # discard the first residual (0 by definition)
+                nonconf_scores = np.full((h,), y[-1])
+                nonconf_scores[: (len(y) - 1)] = y[1:]
+            return nonconf_scores.reshape(1, -1)
+
+        test_length = min(len(y) - 1, h * self.max_num_conformalization_windows)
+        cutoffs = list(range(-h, -test_length - 1, -h))
+
+        nonconf_scores = np.full((len(cutoffs), h), np.nan)
+        for i, cutoff in enumerate(cutoffs, start=0):
+            forecast = self._get_point_forecast(pd.Series(y[:cutoff]), local_model_args)
+            forecast_horizon = y[cutoff:] if cutoff + h == 0 else y[cutoff : cutoff + h]
+            nonconf_scores[i] = np.abs(forecast - forecast_horizon)
+
+        return nonconf_scores
+
+    def _predict_with_local_model(
+        self,
+        time_series: pd.Series,
+        local_model_args: dict,
+    ) -> pd.DataFrame:
+        nonconf_scores = self._get_nonconformity_scores(time_series, local_model_args).ravel()
+
+        # conformalize with naive pooling of nonconformity scores
+        n = len(nonconf_scores)
+        levels = np.array(self.quantile_levels)
+        alpha = 1 - np.abs(2 * levels - 1)  # failure probabilities corresponding to quantiles
+        q_sign = np.sign(2 * levels - 1)
+        ehat = np.quantile(
+            nonconf_scores,
+            q=np.clip(
+                np.ceil((1 - alpha) * (n + 1)) / n,
+                a_min=0.0,
+                a_max=1.0,
+            ),
+            method="lower",
+        )
+
+        point_forecast = self._get_point_forecast(time_series, local_model_args)
+        predictions = {
+            "mean": point_forecast,
+            **{str(q): point_forecast + q_sign[i] * ehat[i] for i, q in enumerate(levels)},
+        }
+        return pd.DataFrame(predictions)
+
+
+class AutoCESModel(AbstractConformalizedStatsForecastModel):
```

Comment:
According to https://nixtla.mintlify.app/statsforecast/src/core/models_intro#automatic-forecasting, `AutoCESModel` is able to produce probabilistic forecasts. Should this model inherit from `AbstractProbabilisticStatsForecastModel` instead?

Reply:
This change would also allow us to remove the `AbstractStatsForecastIntermittentDemandModel` class and make all intermittent models directly inherit from `AbstractConformalizedStatsForecastModel`.

Reply:
I believe this was only added with v1.5.0 to CES.

Reply:
Okay, then we can keep the structure as is.

Can you please add a TODO for this change for when we can finally upgrade to StatsForecast v1.5.0?